build.gradle.kts

```diff
@@ -0,0 +1,153 @@
+import java.net.URI
+import java.time.Instant
+import java.util.Properties
+
+plugins {
+    java
+    `maven-publish`
+    jacoco
+    signing
+    // TODO: static analysis. E.g.:
+    // id("com.diffplug.spotless") version "6.11.0"
+    // id("com.github.spotbugs") version "4.8.0"
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation("junit:junit:4.13.1")
+    testImplementation("org.hamcrest:hamcrest:2.2")
+    testImplementation("pl.pragmatists:JUnitParams:1.1.1")
+}
+
+group = "com.amazon.ion"
+version = "1.9.6-SNAPSHOT"
+description = "A Java implementation of the Amazon Ion data notation."
+// Can't target 6 from 8
+// https://stackoverflow.com/questions/48146930/is-it-possible-to-compile-project-on-java-8-for-target-java-6
+java.sourceCompatibility = JavaVersion.VERSION_1_6
+
+val isReleaseVersion: Boolean = !version.toString().endsWith("SNAPSHOT")
+val generatedJarInfoDir = "${buildDir}/generated/jar-info"
+lateinit var sourcesArtifact: PublishArtifact
+lateinit var javadocArtifact: PublishArtifact
+
+sourceSets {
+    main {
+        java.srcDir("src")
+        resources.srcDir(generatedJarInfoDir)
+    }
+    test {
+        java.srcDir("test")
+    }
+}
+
+tasks {
+    withType<JavaCompile> { options.encoding = "UTF-8" }
+
+    javadoc {
+        // Suppressing Javadoc warnings is clunky, but there doesn't seem to be any nicer way to do it.
+        // https://stackoverflow.com/questions/52205209/configure-gradle-build-to-suppress-javadoc-console-warnings
+        options {
+            (this as CoreJavadocOptions).addStringOption("Xdoclint:none", "-quiet")
+        }
+    }
+
+    create<Jar>("sourcesJar") sourcesJar@ {
+        archiveClassifier.set("sources")
+        from(sourceSets["main"].java.srcDirs)
+        artifacts { sourcesArtifact = archives(this@sourcesJar) }
+    }
+
+    create<Jar>("javadocJar") javadocJar@ {
+        archiveClassifier.set("javadoc")
+        from(javadoc)
+        artifacts { javadocArtifact = archives(this@javadocJar) }
+    }
+
+    val generateJarInfo by creating<Task> {
+        doLast {
+        val propertiesFile = File("$generatedJarInfoDir/${project.name}.properties")
+            propertiesFile.parentFile.mkdirs()
+            val properties = Properties()
+            properties.setProperty("build.version", version.toString())
+            properties.setProperty("build.time", Instant.now().toString())
+            properties.store(propertiesFile.writer(), null)
+        }
+        outputs.dir(generatedJarInfoDir)
+    }
```

Comment:
Please add a comment describing how this is used- by the JarInfo class to allow version etc. introspection at runtime.